pkg/webhook/common.go

```diff
@@ -149,6 +150,22 @@ func (h *webhookHandler) tracingLevel(ctx context.Context, req *admission.Reques
 }
 
 func (h *webhookHandler) skipExcludedNamespace(req *admissionv1.AdmissionRequest, excludedProcess process.Process) (bool, error) {
+	if req.Operation == admissionv1.Delete {
+		// oldObject is the existing object.
+		// It is null for DELETE operations in API servers prior to v1.15.0.
+		// https://github.com/kubernetes/website/pull/14671
+		if req.OldObject.Raw == nil {
+			return false, errOldObjectIsNil
+		}
+
+		// For admission webhooks registered for DELETE operations on k8s built APIs or CRDs,
+		// the apiserver now sends the existing object as admissionRequest.Request.OldObject to the webhook
+		// object is the new object being admitted.
+		// It is null for DELETE operations.
+		// https://github.com/kubernetes/kubernetes/pull/76346
+		req.Object = req.OldObject
```

Comment:
I would recommend setting the req.Object to OldObject, but rather just use `req.OldObject` in local scope when `Object` is nil without setting Object. I understand that down the line we are setting the `req.Object` to `req.OldObject` for delete operations, but I want to make sure that this is only be done at one place rather than two for future purposes, and I also want to be safe in not affecting the code in-between these two points (skipExcludedNamespace, and targetHandler) if it relies on `req.Object`.

Reply:
had some concerns on this as well! no worries I'll change that 👍🏼